Microsoft.Toolkit/Diagnostics/Guard.String.cs

```diff
@@ -0,0 +1,233 @@
+                ThrowHelper.ThrowArgumentExceptionForIsNotNullOrEmpty(text, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input <see cref="string"/> instance must be <see langword="null"/> or whitespace.
+        /// </summary>
+        /// <param name="text">The input <see cref="string"/> instance to test.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="text"/> is neither <see langword="null"/> nor whitespace.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsNullOrWhitespace(string? text, string name)
+        {
+            if (!string.IsNullOrWhiteSpace(text))
+            {
+                ThrowHelper.ThrowArgumentExceptionForIsNullOrWhitespace(text, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input <see cref="string"/> instance must not be <see langword="null"/> or whitespace.
+        /// </summary>
+        /// <param name="text">The input <see cref="string"/> instance to test.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="text"/> is <see langword="null"/> or whitespace.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsNotNullOrWhitespace(string? text, string name)
+        {
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                ThrowHelper.ThrowArgumentExceptionForIsNotNullOrWhitespace(text, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input <see cref="string"/> instance must be empty.
+        /// </summary>
+        /// <param name="text">The input <see cref="string"/> instance to test.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="text"/> is empty.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsEmpty(string text, string name)
+        {
+            if (text.Length != 0)
+            {
+                ThrowHelper.ThrowArgumentExceptionForIsEmpty(text, name);
+            }
+        }
+
+        /// <summary>
+        /// Asserts that the input <see cref="string"/> instance must not be empty.
+        /// </summary>
+        /// <param name="text">The input <see cref="string"/> instance to test.</param>
+        /// <param name="name">The name of the input parameter being tested.</param>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="text"/> is empty.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void IsNotEmpty(string text, string name)
+        {
+            if (text.Length == 0)
```

Comment:
Shouldn't we also be worried about null here?

Reply:
Same as comment above ([link](https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3131#discussion_r390925589)) 👍